python/ray/actor.py

```diff
@@ -265,6 +267,23 @@ def _bind(
         }
         actor._ray_dag_bind_index += 1
 
+        assert (
+            self._signature is not None
+        ), "self._signature should be set for .bind API."
+        try:
+            signature.validate_args(self._signature, args, kwargs)
+        except TypeError as e:
+            signature_copy = self._signature.copy()
+            if len(signature_copy) > 0 and signature_copy[-1].name == "_ray_trace_ctx":
+                # Remove the trace context arg for readability.
+                signature_copy.pop(-1)
+            signature_copy = inspect.Signature(parameters=signature_copy)
+            raise TypeError(
+                f"{str(e)}. The function `{self._method_name}` has a signature "
+                f"`{signature_copy}`, but it doesn't match a given argument to a "
+                f"bind function. args: {args}. kwargs: {kwargs}."
+            ) from None
```

Comment:
I think it should be the other way around:
the bind parameters do not match the method signature

Reply:
f"{str(e)}. The function `{self._method_name}` has a signature "
                f"`{signature_copy}`, but the given arguments to `bind` doesn't "
                f"match. args: {args}. kwargs: {kwargs}."

new error